cli/waiter/token_post.py

```diff
@@ -0,0 +1,126 @@
+import logging
+from enum import Enum
+
+import requests
+
+from waiter import terminal, http_util
+from waiter.querying import get_token
+from waiter.util import FALSE_STRINGS, print_info, response_message, TRUE_STRINGS, guard_no_cluster, str2bool
+
+BOOL_STRINGS = TRUE_STRINGS + FALSE_STRINGS
+
+
+class Action(Enum):
+    CREATE = 1
+    UPDATE = 2
+
+
+def process_post_result(resp):
+    """Prints the result of a token POST"""
+    resp_json = resp.json()
+    if 'message' in resp_json:
+        message = resp_json['message']
+        print_info(f'{message}.')
+        return
+
+    raise Exception(f'{response_message(resp_json)}')
+
+
+def post_failed_message(cluster_name, reason):
+    """Generates a failed token post message with the given cluster name and reason"""
+    return f'Token post {terminal.failed("failed")} on {cluster_name}:\n{terminal.reason(reason)}'
+
+
+def create_or_update(cluster, token_name, token_fields, action):
+    """Creates (or updates) the given token on the given cluster"""
+    cluster_name = cluster['name']
+    cluster_url = cluster['url']
+
+    existing_token_data, existing_token_etag = get_token(cluster, token_name)
+    try:
+        print_info(f'Attempting to {"create" if action == Action.CREATE else "update"} '
+                   f'token on {terminal.bold(cluster_name)}...')
+        json_body = existing_token_data if existing_token_data and action == Action.UPDATE else {}
```

Comment:
Similarly, I would expect this to be a method on the enum and avoid the if-else branching.

Reply:
Done